pkg/controller/nodenetworkconfigurationpolicy/policyconditions/conditions_test.go

```diff
@@ -232,20 +268,25 @@ var _ = Describe("Policy Conditions", func() {
 				e("node1", "policy2", enactmentconditions.SetMatching, enactmentconditions.SetProgressing),
 				e("node2", "policy2", enactmentconditions.SetMatching, enactmentconditions.SetProgressing),
 			},
-			Nodes:  newReadyNodes(3),
+			Nodes:  newNodes(3),
+			Pods:   newNmstatePods(3),
 			Policy: p(setPolicySuccess, "3/3 nodes successfully configured"),
 		}),
-		Entry("when a node is not ready ignore it for policy conditions calculations", ConditionsCase{
+		Entry("when a node does not run nmstate pod ignore it for policy conditions calculations", ConditionsCase{
 			Enactments: []nmstatev1alpha1.NodeNetworkConfigurationEnactment{
 				e("node1", "policy1", enactmentconditions.SetMatching, enactmentconditions.SetSuccess),
 				e("node2", "policy1", enactmentconditions.SetMatching, enactmentconditions.SetSuccess),
 				e("node3", "policy1", enactmentconditions.SetMatching, enactmentconditions.SetSuccess),
 			},
-			Nodes: []corev1.Node{
-				newNode(1, nodeReady()),
-				newNode(2, nodeReady()),
-				newNode(3, nodeReady()),
-				newNode(4, nodeNotReady()),
+			Nodes: newNodes(4),
+			Pods: []corev1.Pod{
+				newNmstatePodAtNode(1),
+				newPodAtNode(1),
+				newNmstatePodAtNode(2),
+				newPodAtNode(2),
+				newNmstatePodAtNode(3),
+				newPodAtNode(3),
```

Comment:
Why do we need so many? Is not one nmstate pod and one non-nmstate enough?

Reply:
To be really sure that we filter nodes with both nmstate and non-nmstate

Reply:
Ack